esp32-wroom-rp/src/wifi.rs

```diff
@@ -40,6 +40,14 @@ where
         Ok(wifi)
     }
 
+    pub(crate) fn build(protocol_handler: NinaProtocolHandler<'a, S, C>) -> Self {
```

Comment:
This allows us to build a new `Wifi` instance without `Tcp` having to know about `WifiCommon`